docs/rfcs/700-config-features-and-defaultpairs-via-connection-string.md

```diff
@@ -0,0 +1,178 @@
+- `value` SHOULD be `true` or `false`.
+
+So a valid connection string containing features could be:
+
+- `s3://bucket_name/prefix?credential=hmac:xxxx:xxxx&enable_loose_pair=false&enable_virtual_dir=true`
+
+The connection string above is equivalent to:
+
+```go
+store, err := s3.NewStorage(
+	ps.WithCredential("hmac:xxxx:xxxx"),
+	ps.WithName("bucket_name"),
+	ps.WithWorkDir("/prefix"),
+	s3.WithStorageFeatures(s3.StorageFeaturs{
+		LoosePair:  false,
+		VirtualDir: true,
+    })
+
+)
+```
+
+### Default Pairs
+
+The format of default pair in connection string is:
+
+`default_key=value`
+
+- The whole key of the pair SHOULD have the prefix `default_`.
+- `key` is the global or service pair name defined in `toml`, and the format SHOULD be exactly the same.
+- For `value`, parsable value types are `string`, `bool`, `int`, `int64`, `uint64`, `[]byte` and `time.Duration` at present.
+- Pairs of operation with the same name will share the default value.
+
+A valid connection string containing default pairs could be:
+
+- `s3://bucket_name/prefix?credential=hmac:xxxx:xxxx&default_server_side_encryption=xxxx&default_strorage_class=xxxx`
+
+Also, we generate the following function to support configure the default pair during initialization in the way of `WithXxx()`:
+
+Take `default_storage_class` for example:
+
+```go
+func WithDefaultStorageClass(v string) Pair {
+	reutrn Pair{
+		Key:   "default_storage_class",
+		Value: v,
+	}
+}
+```
+
+The connection string above is equivalent to:
+
+```go
+store, err := s3.NewStorage(
+	ps.WithCredential("hmac:xxxx:xxxx"),
+	ps.WithName("bucket_name"),
+	ps.WithWorkDir("/prefix"),
+    }), 
+    s3.WithDefaultServiceSideEncryption("xxxx"),
+    s3.WithDefaultStorageClass("xxxx"),
+)
+```
+
+That means pair of operations with the same name `server_side_encryption` or `storage_class` will share the default value.
+
+### Implementation
+
+ Default pairs and features are defined in service.
+
+**Add pairs for features and default paris**
+
+Default pairs and features can be added into `Optioanl` pairs of `New` for service, like:
+
+```toml
+[namespace.storage.new]
+required = ["name"]
+optional = ["storage_features", "default_storage_pairs", "default_storage_class", "enable_virtual_dir"]
+```
+
+When parsing pairs from toml, we should:
+
+- Check whether the default pair belongs to global or system pair and feature belongs to the supported ones.
+- Add pairs for the optional pairs of `New` with `default_` and `enable_` prefix.
+- Mark them as defaultable, the same with global and system paris corresponding to the default ones. 
+
+**Parse features in `parsePair*New`**
+
+We can handle the optional defaultable pairs with prefix `enable_`, assign value to the corresponding field of `*Features` in `ParsePair*New()`.
+
+**Parse default pairs in `parsePair*New`**
+
+We can add private fields correspond to default pairs into the generated `Default*Pairs` to carry the shared values, like:

```

Comment:
Maybe we can convert `default_storage_class=xxxx`/`WithStoragClass("xxx")` to pairs in `DefaultStoragePairs` directly?

- We know storage_class is a defaultable pair
- We know which operations support storage_class